ui/src/views/explore/list-nfas/nfa-list/nfa-list.tsx

```diff
@@ -43,7 +47,8 @@ export const NFAList: React.FC = () => {
 
   useWindowScrollEnd(() => {
     if (isLoading || endReached || queryError) return;
-    setPageNumber((prevState) => prevState + 1);
+    const newPageNumber = pageNumber + 1;
+    setPageNumber(newPageNumber);

```

Comment:
There is any special reason for this change?
```suggestion
    setPageNumber(pageNumber + 1);
```